### PR TITLE
Fix steps_to_mm for active extruders

### PR DESCRIPTION
**Pull request by MagoKimbra**
In LOOP_XYZE steps_to_mm must to set to active_extruders...

### CODE REVIEW

**Comment by thinkyhead** [comment]
This probably addresses some recent issues. Here's hoping!

File: Marlin/planner.cpp [hunk 1]
```diff
@@ -1418,7 +1418,7 @@ void Planner::set_position_mm_kinematic(const float position[NUM_AXIS]) {
 void Planner::sync_from_steppers() {
   LOOP_XYZE(i) position[i] = stepper.position((AxisEnum)i);
   #if ENABLED(LIN_ADVANCE)
-    LOOP_XYZE(i) position_float[i] = stepper.position((AxisEnum)i) * steps_to_mm[i];
+    LOOP_XYZE(i) position_float[i] = stepper.position((AxisEnum)i) * (i == E_AXIS ? steps_to_mm[E_AXIS + active_extruder] : steps_to_mm[i]);
```
Comment (thinkyhead) [inline Marlin/planner.cpp:1421]: …or…
```cpp
LOOP_XYZE(i) position_float[i] = stepper.position((AxisEnum)i) * steps_to_mm[i + (i == E_AXIS ? active_extruder : 0)];
```
😁

Comment (MagoKimbra) [reply Marlin/planner.cpp:1421]: Yes! :)

Comment (thinkyhead) [reply Marlin/planner.cpp:1421]: Or, actually, I forgot this is optional behavior… https://github.com/MarlinFirmware/Marlin/pull/5855/files

**Comment by MagoKimbra** [comment]
Ops sorry i do not remember that you use DISTINCT_E_FACTORS option...

**Comment by thinkyhead** [comment]
All these conditional compiles make the head spin. But they do make it easy to pare down to something very minimal, like this i2c z-axis slave controller made from Marlin… https://github.com/thinkyhead/Marlin/tree/a3d_zaxis_controller/Marlin

**Pull request merged**